File: php/src/Model/SupervisorSession.php
<?php

declare(strict_types=1);

namespace Fivexer\SDK\Model;

/**
 * A redeemed supervisor session. `expiresAt` is epoch-**milliseconds**, not the ISO-8601 string the worker plane sends — the two planes genuinely differ on the wire, and this mirrors the server rather than papering over it.
 *
 * Response model: built from a decoded JSON array and read-only thereafter.
 */
final class SupervisorSession
{
    public function __construct(
        /** An `sv_` supervisor session token. */
        public readonly string $token,
        /** Epoch-milliseconds. */
        public readonly int $expiresAt,
        public readonly SupervisorIdentity $supervisor,
        public readonly string $workspaceId,
    ) {
    }

    /**
     * @param array<string, mixed> $data
     * @internal
     */
    public static function fromArray(array $data): self
    {
        return new self(
            token: (string) ($data['token'] ?? ''),
            expiresAt: (int) ($data['expiresAt'] ?? 0),
            supervisor: SupervisorIdentity::fromArray($data['supervisor'] ?? ['id' => '']),
            workspaceId: (string) ($data['workspaceId'] ?? ''),
        );
    }
}

File: java/src/main/java/io/fivexer/sdk/model/SupervisorSession.java
package io.fivexer.sdk.model;

/**
 * A redeemed supervisor session. {@code expiresAt} is epoch-<em>milliseconds</em>, not the ISO-8601 string the worker plane sends — the two planes genuinely differ on the wire, and this mirrors the server rather than papering over it.
 *
 * <p>Populated by Gson via field reflection and exposed read-only; callers receive instances
 * from the client and never construct them.
 */
public class SupervisorSession {
    private String token;
    private Long expiresAt;
    private SupervisorIdentity supervisor;
    private String workspaceId;

    public String getToken() { return token; }
    public Long getExpiresAt() { return expiresAt; }
    public SupervisorIdentity getSupervisor() { return supervisor; }
    public String getWorkspaceId() { return workspaceId; }
}
